src/engine/Board.java

```diff
@@ -484,23 +486,44 @@ public Zombie getSingleAirTarget(int x, int y) {
 	 * Deletes all the units within the specified row
 	 * @param row - the row that all the units will be deleted in
 	 */
-	public void useLawnMower(int row)
+	public List<Zombie> useLawnMower(int row)
 	{
-		for(Unit u : getRowUnits(row))
+		List<Unit> unitRemoveBin = getRowUnits(row);
+		List<Zombie> zomRemoveBin = new LinkedList<Zombie>();
+		
+		if(!unitRemoveBin.isEmpty())
 		{
-			if(u instanceof Zombie)
-			{
-				removeZombie(u.getRow(),u.getCol());
-			}
-			else if(u instanceof Plant)
+			for(Unit u : unitRemoveBin)
 			{
-				removePlant(u.getRow(),u.getCol());
+				if(u instanceof Zombie)
+				{
+					removeZombie(u.getRow(),u.getCol());
+					zomRemoveBin.add((Zombie) u);
+					LOG.debug("Lawnmower kills Zombie " + u.getRow() + " " + u.getCol());
+				}
+				else if(u instanceof Plant)
+				{
+					removePlant(u.getRow(),u.getCol());
+					LOG.debug("Lawnmower kills Plant " + u.getRow() + " " + u.getCol());
+				}
 			}
 		}
+		return zomRemoveBin;
+	}
+	
+	/**
+	 * Checks if the mower is avaliable for the given row
+	 * @param row - the row to check if the lawn mower is avaliable
+	 * @return true if the lawn mower is avaliable for use, otherwise return false
+	 */
+	public boolean isMowerAvaliable(int row)
+	{
+		LOG.debug("Checking if lawn mower " + row + " is available and it returns " + mowersAvaliable.contains(row));
+		return mowersAvaliable.contains(row);
 	}
 	
 	@Override
-	public boolean onZombieMove(Zombie zombie, int maxRow) {
+	public boolean onZombieMove(Zombie zombie, Game game) {
```

Comment:
Instead of passing a `Game` object to this method, maybe check for lawn mower use inside `zombieTurn()` method?

We are already checking if a zombie has reached the end in that method anyways.

Reply:
This won't be possible. Game is passed for two reasons, one to process the new zombie location after a move (via getNewZomPosition) and two, to update the lawn mower in the view (via GameListener). The problem is that to update the lawn mower in view, the row of the lawn mower is needed. I can't return it back to Game because it is already used to return a boolean value, thus it seems I must send the Game instance to the board.